src/main/java/org/openstreetmap/atlas/geography/sharding/AppendCountsForAllZoom.java

```diff
@@ -0,0 +1,179 @@
+package org.openstreetmap.atlas.geography.sharding;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
+import org.openstreetmap.atlas.streaming.Streams;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.streaming.resource.WritableResource;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.openstreetmap.atlas.utilities.runtime.Command;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Command that takes a file with feature counts for each {@link SlippyTile} at a certain zoom
+ * level, and then calculates summed counts for every lower zoom layer to the file. This is meant to
+ * be used as a pre-processing step before creating the sharding tree in
+ * {@link DynamicTileSharding}.
+ *
+ * @author james-gage
+ * @author lcram
+ */
+public class AppendCountsForAllZoom extends Command
+{
+
+    public static final Switch<WritableResource> DEFINITION = new Switch<>("definition",
+            "Resource containing tile to feature count mapping for zoom 0 to maxZoom-1.", File::new,
+            Optionality.REQUIRED);
+    public static final Switch<WritableResource> OUTPUT = new Switch<>("output",
+            "The resource where to save the feature counts for all zooms.", File::new,
+            Optionality.REQUIRED);
+    private static final Switch<Integer> ZOOM = new Switch<>("zoom",
+            "The zoom layer the input csv represents", Integer::valueOf, Optionality.REQUIRED);
```

Comment:
Nit: from this description, it's a little unclear what the CSV input is - is it the definition file? If so, can you add that to the description please?